crates/red_knot_python_semantic/src/types.rs

```diff
@@ -467,6 +467,16 @@ impl<'db> Type<'db> {
             {
                 true
             }
+            (Type::Tuple(self_tuple), Type::Tuple(target_tuple)) => {
+                self_tuple.len(db) == target_tuple.len(db)
+                    && self_tuple
+                        .elements(db)
+                        .iter()
+                        .zip(target_tuple.elements(db))
+                        .all(|(self_element, target_element)| {
+                            self_element.is_subtype_of(db, *target_element)
+                        })
```

Comment:
Nit: Not sure if Rustc can deduplicate the `self_tuple.elements(db).len()` (inlined `len`) and the `self_tuple(db)` because of the dyn `Db`. 

```suggestion
								let self_elements = self_tuple.elements(db);
								let target_elements = target_tuple.elements(db);
								
                self_elements.len(db) == target_elements.len(db)
                    && self_elements
                        .iter()
                        .zip(target_elements)
                        .all(|(self_element, target_element)| {
                            self_element.is_subtype_of(db, *target_element)
                        })
```